packages/davinci-client/src/lib/client.store.ts

```diff
@@ -190,11 +206,24 @@ export async function davinci<ActionType extends ActionTypes = ActionTypes>({
       }
 
       const { id } = collector;
-      const collectorToUpdate = nodeSlice.selectors.selectCollector(store.getState(), id);
+      const { error, state: collectorToUpdate } = nodeSlice.selectors.selectCollector(
+        store.getState(),
+        id,
+      );
+
+      if (error) {
+        log.error(error.message);
```

Comment:
Curious what your thoughts on this are as it relates to logging the error, and then returning an error value.

I see why we are doing it, but I'm wondering if the API is a bit odd to see an error but then have a function (which also errors).